src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,171 @@
+    }
+    // Call the submit handler for the settings component to save its settings
+    if (onSettingsSave) {
+      onSettingsSave();
+    }
+  };
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL} target="_blank">{learnMoreText}</Hyperlink>
+  );
+
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className="bg-white d-flex flex-column mw-xs p-3"
+      >
+        {
+          loadingStatus === RequestStatus.SUCCESSFUL && (
+            <Formik
+              initialValues={{
+                enabled: !!appInfo?.enabled,
+                ...initialValues,
+              }}
+              validationSchema={
+                Yup.object()
+                  .shape({
+                    enabled: Yup.boolean(),
+                    ...validationSchema,
+                  })
+              }
+              onSubmit={handleFormSubmit}
+            >{(formikProps) => (
+              <Form onSubmit={formikProps.handleSubmit}>
+                <h3>{title}</h3>
+                <AppConfigFormDivider />
+                <FormSwitchGroup
+                  id={`enable-${appId}-toggle`}
+                  name="enabled"
+                  onChange={formikProps.handleChange}
+                  onBlur={formikProps.handleBlur}
+                  checked={formikProps.values.enabled}
+                  label={(
+                    <>
+                      {enableAppLabel}&nbsp;
+                      <StatusBadge loadingStatus={formikProps.values.enabled} />
+                    </>
+                  )}
+                  helpText={(<p>{enableAppHelp}<br />{learnMoreLink}</p>)}
+                />
+                <AppConfigFormDivider />
+                <AppSettingsForm formikProps={formikProps}>
+                  {children}
+                </AppSettingsForm>
+                {formikProps.values.enabled && children && <AppConfigFormDivider />}
+                <div className="d-flex justify-content-end">
+                  <Button variant="cancel" className="btn btn-link" onClick={onClose}>
+                    {intl.formatMessage(messages.cancel)}
+                  </Button>
+                  <Button type="submit" variant="success" data-autofocus>
+                    {intl.formatMessage(messages.apply)}
+                  </Button>
+                </div>
+              </Form>
+            )}
+            </Formik>
+          )
+        }
+        {loadingStatus === RequestStatus.IN_PROGRESS && (
+          <Spinner animation="border" variant="primary" className="align-self-center" />
+        )}
+      </div>
+    </ModalLayer>
+  );
+}
+
+AppSettingsModal.propTypes = {
+  intl: intlShape.isRequired,
+  title: PropTypes.string.isRequired,
+  appId: PropTypes.string.isRequired,
+  children: PropTypes.func,
```

Comment:
are you expecting `children` to be func?

Reply:
Yes.

Reply:
I saw you are using `{children}` above, is that expected? If its a function you should be calling that no? `{children()}`

Reply:
Since this code was getting pretty convoluted I broke it into smaller components in the same file. So here I'm passing the function to AppSettingsForm as-is, and AppSettingsForm is then passing calling the function and passing it props from formik